clients/da-vinci-client/src/main/java/com/linkedin/davinci/blobtransfer/BlobSnapshotManager.java

```diff
@@ -117,12 +117,10 @@ public BlobTransferPartitionMetadata getTransferMetadata(BlobTransferPayload pay
 
     // check if the concurrent user count exceeds the limit
     checkIfConcurrentUserExceedsLimit(topicName, partitionId);
-    concurrentSnapshotUsers.computeIfAbsent(topicName, k -> new VeniceConcurrentHashMap<>())
-        .computeIfAbsent(partitionId, k -> new AtomicLong(0))
-        .incrementAndGet();
 
     boolean isHybrid = isStoreHybrid(payload.getStoreName());
```

Comment:
BTW,
Can this function be invoked concurrently?
If yes, we may need to make it `synchronized` to avoid the race condition.

Reply:
Yes, it could be called when there are several requests at the same time.

Reply:
So shall we make it thread-safe to avoid creating multiple snapshots?

Reply:
Sure, added synchronized.